coordinator/src/main/scala/com/socrata/datacoordinator/service/collocation/CollocationLock.scala

```diff
@@ -0,0 +1,37 @@
+package com.socrata.datacoordinator.service.collocation
+
+import java.util.concurrent.TimeUnit
+
+import org.apache.curator.framework.CuratorFramework
+import org.apache.curator.framework.recipes.locks.InterProcessMutex
+
+sealed abstract class CollocationLockException(message: String, cause: Exception) extends Exception(message, cause)
+
+case class CollocationLockTimeout(millis: Long)
+  extends CollocationLockException(s"Timeout while waiting to acquire collocation lock after $millis milliseconds", null)
+case class CollocationLockError(action: String, cause: Exception)
+  extends CollocationLockException(s"Unexpected exception while $action collocation lock", cause)
+
+trait CollocationLock {
+  def acquire(timeoutMillis: Long): Option[CollocationLockTimeout]
+  def release(): Unit
+}
+
+class CuratedCollocationLock(curator: CuratorFramework) extends CollocationLock {
+  // TODO: what is the right type of curator lock thing to use here?
+  val lock = new InterProcessMutex(curator, "/com.socrata/soda/data-coordinator/collocation-lock")
```

Comment:
Hardcoding this feels a bit icky ... wonder if using the `curator.namespace` config makes sense.

Reply:
This seems like the right curator lock, as you need a timeout.